databricks/resource_databricks_permissions.go

```diff
@@ -0,0 +1,209 @@
+	id := d.Id()
+	client := m.(*service.DBApiClient)
+	err := client.Permissions().SetOrDelete(id, new(model.AccessControlChangeList))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// permissionsIDFieldMapping holds mapping
+type permissionsIDFieldMapping struct {
+	field        string
+	objectType   string
+	resourceType string
+	idRetriever  func(client *service.DBApiClient, id string) (string, error)
+}
+
+// PermissionsResourceIDFields shows mapping of id columns to resource types
+func permissionsResourceIDFields() []permissionsIDFieldMapping {
+	SIMPLE := func(client *service.DBApiClient, id string) (string, error) {
+		return id, nil
+	}
+	PATH := func(client *service.DBApiClient, path string) (string, error) {
+		info, err := client.Notebooks().Read(path)
+		if err != nil {
+			return "", errors.Wrapf(err, "Cannot load path %s", path)
+		}
+		return strconv.FormatInt(info.ObjectID, 10), nil
+	}
+	return []permissionsIDFieldMapping{
+		{"cluster_policy_id", "cluster-policy", "cluster-policies", SIMPLE},
+		{"instance_pool_id", "instance-pool", "instance-pools", SIMPLE},
+		{"cluster_id", "cluster", "clusters", SIMPLE},
+		{"job_id", "job", "jobs", SIMPLE},
+		{"notebook_id", "notebook", "notebooks", SIMPLE},
+		{"notebook_path", "notebook", "notebooks", PATH},
+		{"directory_id", "directory", "directories", SIMPLE},
+		{"directory_path", "directory", "directories", PATH},
+	}
+}
+
+func conflictingFields(field string) []string {
+	conflicting := []string{}
+	for _, mapping := range permissionsResourceIDFields() {
+		if mapping.field == field {
+			continue
+		}
+		conflicting = append(conflicting, mapping.field)
+	}
+	return conflicting
+}
+
+func resourcePermissions() *schema.Resource {
+	fields := map[string]*schema.Schema{
+		"object_type": {
+			Type:     schema.TypeString,
+			Computed: true,
+		},
+		"access_control": {
+			ForceNew:   true,
+			Type:       schema.TypeList,
+			MinItems:   1,
+			Required:   true,
+			ConfigMode: schema.SchemaConfigModeAttr,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"user_name": {
+						ForceNew: true,
+						Type:     schema.TypeString,
+						Optional: true,
+						//ConflictsWith: []string{"group_name"},
+					},
+					"group_name": {
+						ForceNew: true,
+						Type:     schema.TypeString,
+						Optional: true,
+						//ConflictsWith: []string{"user_name"},
+					},
+					"permission_level": {
+						ForceNew: true,
+						Type:     schema.TypeString,
+						Required: true,
+					},
+				},
+			},
+		},
+	}
+	for _, mapping := range permissionsResourceIDFields() {
+		fields[mapping.field] = &schema.Schema{
+			ForceNew:      true,
+			Type:          schema.TypeString,
+			Optional:      true,
+			ConflictsWith: conflictingFields(mapping.field),
+		}
+	}
+	return &schema.Resource{
+		Create: resourcePermissionsCreate,
+		Read:   resourcePermissionsRead,
+		Delete: resourcePermissionsDelete,
```

Comment:
Is there a specific reason for not having an update? It seems as though the apis support patches. Is it for simplicity's sake? It seems that there is both a PUT (create/delete) and a PATCH (update) operations available.

Reply:
per se, permissions are not deletable... there'll always be inherited permission for the admin creating the resource